packages/generator/src/shared/wirePropertyV2ToSdkPrimaryKeyTypeDefinition.ts

```diff
@@ -23,13 +23,33 @@ export function wirePropertyV2ToSdkPrimaryKeyTypeDefinition(
   switch (input.dataType.type) {
     case "integer":
     case "double":
-    case "string": {
+    case "string":
+    case "boolean":
+    case "attachment":
+    case "byte":
+    case "decimal":
+    case "float":
+    case "geopoint":
+    case "geoshape":
+    case "long":
+    case "short": {
       return input.dataType.type;
     }
-    default: {
+    case "date": {
+      return "datetime";
```

Comment:
```suggestion
      return "localdate";
```
may require a bunch of downstream changes though

Reply:
Lets revisit this one, we can also rename it to be consistent with the type in the ontology metadata (just `date`)